rope/contrib/autoimport/sqlite.py

```diff
@@ -153,7 +153,10 @@ def create_database_connection(
             raise Exception("if memory=False, project must be provided")
         db_path: str
         if memory or project is None or project.ropefolder is None:
-            db_path = ":memory:"
+            # Allows the in-memory db to be shared across threads
+            # See https://www.sqlite.org/inmemorydb.html
+            project_hash = hash(project and project.ropefolder and project.ropefolder.real_path)
+            db_path = f"file:memdb{project_hash}:?mode=memory&cache=shared"
```

Comment:
This creates a database file, i.e. writes everything to disk, even though I followed https://www.sqlite.org/inmemorydb.html!